Add links for product ingested traces (#13830)
DataDog/documentation: content/en/tracing/trace_ingestion/mechanisms.md

Some additional ingestion reasons are attributed to spans that are generated by specific Datadog products:

| Product    | Ingestion Reason                    | Ingestion Mechanism Description |
|------------|-------------------------------------|---------------------------------|

> | Synthetic Monitoring | `synthetics` and `synthetics-browser` | An HTTP or browser test generates a trace when the backend services are instrumented. [Find the backend trace][4] from the synthetic test run.

Reviewer: ```suggestion
| Synthetic Monitoring | `synthetics` and `synthetics-browser` | An HTTP or browser test generates a trace when the backend services are instrumented. Find the backend trace from the [synthetic test][4] run.
```

Reviewer: Small update to make the link text more descriptive.

> | Real User Monitoring        | `RUM`                               | A browser request from a web or mobile application generates a trace when the backend services are instrumented. [Find the backend trace][5] from the RUM browser sessions and resources. |

Reviewer: ```suggestion
| Real User Monitoring        | `RUM`                               | A browser request from a web or mobile application generates a trace when the backend services are instrumented. Find the backend trace from the [RUM browser sessions][5] and resources. |
```

Reviewer: Small update to make the link text more descriptive.

> | Serverless | `lambda` and `xray`                   | Your traces received from the [Datadog Lambda Extension][6] or AWS X-Ray integration |

Author: ```suggestion
| Serverless | `lambda` and `xray`                   | Your traces received from the [Serverless applications][6] traced with Datadog Tracing Libraries or the AWS X-Ray integration |
```

> | Serverless | `lambda` and `xray`                   | Your traces received from the [Serverless applications][6] traced with Datadog Tracing Libraries or the AWS X-Ray integration |
> | Application Security Monitoring     | `appsec`                            | Traces ingested from Datadog tracing libraries and flagged by [ASM][7] as a threat |

Author: ```suggestion
| Serverless | `lambda` and `xray`                   | Your traces received from the [Serverless applications][6] traced with Datadog Tracing Libraries or the AWS X-Ray integration. |
| Application Security Monitoring     | `appsec`                            | Traces ingested from Datadog tracing libraries and flagged by [ASM][7] as a threat. |
```

[1]: /tracing/setup_overview/setup/
[2]: /tracing/visualization/#trace-root-span
[3]: /tracing/trace_ingestion/control_page
[4]: /synthetics/apm/
[5]: /real_user_monitoring/connect_rum_and_traces/

> [6]: /serverless/distributed_tracing/

Reviewer: ```suggestion
[6]: /serverless/libraries_integrations/extension/
```

Reviewer: Is this link intended to point to the Lambda extension, as the link text on line 199 would suggest? If the link is indeed meant for /serverless/distributed_tracing/, could you update the link text to "serverless applications" or something similar?
